fs/zfs broken on FreeBSD prior to release 13

FreeBSD versions prior to release 13 did not use OpenZFS as the ZFS implementation. As a result, d6dab2c broke the build of fs/zfs on FreeBSD 12 and earlier.

```
vdev_status.c:29:5: error: unknown type name 'zpool_errata_t'; did you mean 'zpool_data_t'?
    zpool_errata_t errata;
    ^~~~~~~~~~~~~~
    zpool_data_t
./config.h:59:3: note: 'zpool_data_t' declared here
} zpool_data_t;
  ^
vdev_status.c:30:44: error: too many arguments to function call, expected 2, have 3
    reason = zpool_get_status(zhp, &msgId, &errata);
             ~~~~~~~~~~~~~~~~              ^~~~~~~
/usr/src/cddl/contrib/opensolaris/lib/libzfs/common/libzfs.h:377:1: note: 'zpool_get_status' declared here
extern zpool_status_t zpool_get_status(zpool_handle_t *, char **);
^
vdev_status.c:242:18: error: use of undeclared identifier 'ZPOOL_STATUS_ERRATA'; did you mean 'ZPOOL_STATUS_OK'?
            case ZPOOL_STATUS_ERRATA:
                 ^~~~~~~~~~~~~~~~~~~
                 ZPOOL_STATUS_OK
/usr/src/cddl/contrib/opensolaris/lib/libzfs/common/libzfs.h:374:2: note: 'ZPOOL_STATUS_OK' declared here
        ZPOOL_STATUS_OK
        ^
vdev_status.c:251:18: error: duplicate case value 'ZPOOL_STATUS_OK'
            case ZPOOL_STATUS_OK:
                 ^
vdev_status.c:242:18: note: previous case defined here
            case ZPOOL_STATUS_ERRATA:
                 ^
4 errors generated.
gmake: *** [<builtin>: vdev_status.o] Error 1
```

--- a/modules/fs/zfs/src/Makefile
+++ b/modules/fs/zfs/src/Makefile
@@ -40,6 +40,10 @@ LDADD= 		-lzfs -lnvpair
 
 endif 
 
+ifeq  ($(and $(shell test $(UNAME_S) == "FreeBSD"; echo $$?),$(shell test `uname -K` -lt 1300000; echo $$?)),0)
+CFLAGS+=	-DFREEBSD_NO_OPENZFS
+endif
+
 all: zio.c $(OBJ)
 	$(CC) $(CFLAGS) $(DPADD) -o $(PROG) $(OBJ) $(LDADD)
 

--- a/modules/fs/zfs/src/vdev_status.c
+++ b/modules/fs/zfs/src/vdev_status.c
@@ -23,8 +23,12 @@ zpool_print_vdev(zpool_handle_t * zhp, void * data) {
     nvlist_t * config, * nvroot;
     vdev_stat_t * vs;
 
+#if defined(FREEBSD_NO_OPENZFS)
+    reason = zpool_get_status(zhp, &msgId);
+#else
     zpool_errata_t errata;
     reason = zpool_get_status(zhp, &msgId, &errata);
+#endif
 
     config = zpool_get_config(zhp, NULL);
 
@@ -234,6 +238,7 @@ zpool_print_vdev(zpool_handle_t * zhp, void * data) {
                         "\tThen import it to correct the mismatch.\n");
                 break;
 
+#if !defined(FREEBSD_NO_OPENZFS)
             case ZPOOL_STATUS_ERRATA:
                 (void) snprintf(zstat->err_message, ERR_MESSAGE_SIZE, "status: The ZFS pool contains an on-disk "
                         "format incompatibility.\n"
@@ -241,6 +246,7 @@ zpool_print_vdev(zpool_handle_t * zhp, void * data) {
                         "This will return the pool to a state in which it may be imported by other implementations.\n"
                         "action: Scrub the affected pool with 'zpool scrub'.\n");
                 break;
+#endif
 
             case ZPOOL_STATUS_OK:
                 (void) snprintf(zstat->err_message, ERR_MESSAGE_SIZE, "status: OK\n");

